developer-hub/src/reference_docs/glossary.rs

```diff
@@ -36,41 +36,71 @@
 //!
 //! #### Offchain
 //!
+//! Refers to operations conducted outside the blockchain's consensus mechanism. They are essential
+//! for enhancing scalability and efficiency, enabling activities like data fetching and computation
+//! without bloating the blockchain.
+//!
 //! #### Host Function:
 //!
+//! Host functions are functions provided by the runtime environment (the [host](#host)) to the Wasm
+//! runtime. These functions allow the Wasm code to interact with and perform operations on the
+//! [node](#node), like accessing the blockchain state.
+//!
 //! #### Runtime API:
 //!
+//! The runtime API acts as a communication bridge between the runtime and the node, serving as the
+//! exposed interface that facilitates their interactions.
+//!
 //! #### Dispatchable:
 //!
-//! Callable
+//! Dispatchables are functions that can be called by external entities, such as users or external
+//! systems, to interact with the blockchain's state. They are a core aspect of the runtime logic,
+//! handling transactions and other state-changing operations.
+//!
+//! **Synonyms**: Callable
 //!
 //! #### Extrinsic
 //!
+//! An extrinsic is a general term for a piece of data that is originated outside of the runtime and
+//! fed into the it as a part of the block-body. This includes user-initiated transactions as well
+//! as inherents which are placed into the block by the block-builder.
 //!
 //! #### Pallet
 //!
-//! #### Full Node
-//!
-//! #### Archive Node
-//!
-//! #### Validator
-//!
-//! #### Collator
-//!
-//! #### Parachain
-//!
-//! aka. AppChain.
-//!
-//!
-//!
+//! FRAME pallets are modular components that encapsulate specific functionalities or
+//! business logic of a blockchain. They are the building blocks used to construct a blockchain's
+//! runtime, allowing for customizable and upgradeable networks. Pallets can be used to extend the
+//! capabilities of a Substrate-based blockchain in a composable way.
```

Comment:
Do we have a link to a more in-depth pallet explanation?

Reply:
We have this https://docs.substrate.io/reference/frame-pallets/, that we don't want to link to as the idea is to deprecate it. Maybe we can add a 'pallet' section to the reference docs with that content...
@CrackTheCode016 I know that you want to port over some of the content from docs.substrate.io, maybe that's something worth migrating, wdyt?

Reply:
this is a great point, i think that (and other things like the SCALE stuff) could make it in the ref docs. will consider when adding to the issue

Reply:
and if not in the ref docs, then it can even have its own entry point maybe where it makes sense

Reply:
@CrackTheCode016 I have created a task for this so we don't forget it https://github.com/paritytech/polkadot-sdk-docs/issues/61